src/sonic-bgpcfgd/bgpcfgd/main.py

```diff
@@ -96,6 +95,12 @@ def do_work():
         log_notice("software_bfd feature is enabled, starting bfd manager")
         managers.append(BfdMgr(common_objs, "STATE_DB", swsscommon.STATE_BFD_SOFTWARE_SESSION_TABLE_NAME))
 
+    device_metadata = config_db.get_table("DEVICE_METADATA")
+    if "localhost" in device_metadata and "type" in device_metadata["localhost"] and device_metadata["localhost"]["type"] == "SpineRouter" and "subtype" in device_metadata["localhost"] and device_metadata["localhost"]["subtype"] == "UpstreamLC":
```

Comment:
Do we need consider disaggregated t2 here?

Reply:
have sync offline, will support it later in other pr